### PR TITLE
com.abagames.noiz2sa

### PR DESCRIPTION
**Pull request by joaquimrocha**
Hi, I had previously opened [PR#633](https://github.com/flathub/flathub/pull/633) , but somehow screwed up when I force-pushed the branch and this resulted in the PR#633 getting closed (and I cannot reopen it).

So here a new PR. I have addressed a few changes that @TingPing had indicated. I have also tried to use the suggested makefile-related manifest changes for `bulletml` module, but couldn't really make it work in the time I have. So maybe let's keep the current build-commands.

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build com.abagames.noiz2sa

File: com.abagames.noiz2sa.json [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    "--socket=pulseaudio",
+    "--socket=x11",
+    "--persist=.noiz2sa.prf"
+  ],
+  "modules": [
+    "shared-modules/glu/glu-9.0.0.json",
+    "shared-modules/SDL/SDL-1.2.15.json",
+    "shared-modules/SDL/SDL_mixer-1.2.12.json",
+    {
+      "name": "bulletml",
```
Comment (TingPing) [inline com.abagames.noiz2sa.json]: Add `cleanup` to remove headers, etc.

File: com.abagames.noiz2sa.json [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+          "path": "patches/bulletml/includes.patch"
+        },
+        {
+          "type": "patch",
+          "path": "patches/bulletml/get-rid-of-boost.patch"
+        }
+      ],
+      "build-commands": [
+        "rm -f src/calc.cpp",
+        "rm -f src/*.o src/*/*.o",
```
Comment (TingPing) [inline com.abagames.noiz2sa.json]: The source tarball contains these?!

File: com.abagames.noiz2sa.json [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        "make -C src CXXFLAGS=\"$(CXXFLAGS) $(CPPFLAGS)\" libbulletml.a",
+        "rm -f src/*.o src/*/*.o",
```
Comment (TingPing) [inline com.abagames.noiz2sa.json]: `make clean` maybe?

File: com.abagames.noiz2sa.json (hunk 2 above)
Comment (joaquimrocha) [reply com.abagames.noiz2sa.json]: I copied these instructions from the Debian package's rules without many changes. I will double-check this.

File: com.abagames.noiz2sa.json (hunk 3 above)
Comment (joaquimrocha) [reply com.abagames.noiz2sa.json]: Again, I copied these instructions from Debian, but apparently `make clean` does the same...

**Comment by joaquimrocha** [comment]
@TingPing I have simplified the `bulletml` recipe. Hopefully is good to go now.

**Comment by nedrichards** [comment]
bot, build com.abagames.noiz2sa

File: com.abagames.noiz2sa.json [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+          "type": "shell",
+          "commands": [
+            "install -Dm644 noiz2sa.appdata.xml /app/share/appdata/noiz2sa.appdata.xml"
+          ]
+        }
+      ],
+      "build-commands": [
+        "install ./noiz2sa /app/bin/noiz2sa",
+        "install -d /app/share/games/noiz2sa",
+        "cp -R ../noiz2sa_share/* /app/share/games/noiz2sa"
```
Comment (TingPing) [inline com.abagames.noiz2sa.json]: Can combine these:

```
install -m644 --target-directory /app/share/games/noiz2sa ../noiz2sa_share/*
```

Comment (joaquimrocha) [reply com.abagames.noiz2sa.json]: There are many manifest copying files in different ways. I don't think such changes should block PRs. I will update the manifest to do as you said, but let's not block things on nitpicks like these please, as any such improvements can come in later commits too.

Comment (joaquimrocha) [reply com.abagames.noiz2sa.json]: And in fact your suggestion seems to fail:
```
Running: install ./noiz2sa /app/bin/noiz2sa
Running: install -m644 --target-directory=/app/share/games/noiz2sa ../noiz2sa_share/*
install: failed to access '/app/share/games/noiz2sa': No such file or directory
```

So I will leave it as is.

**Comment by barthalion** [comment]
bot, build com.abagames.noiz2sa

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/com.abagames.noiz2sa

Thanks!

**Pull request closed without merging**